File: 1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start


class Solution:
    def twoSum(self, nums, target: int):
        for i in range(0, len(nums)):
            for j in range(i+1, len(nums)):
                if nums[i]+nums[j] == target:
                    return [i, j]


class Solution:
    def twoSum(self, nums, target: int):
        o = {}
        for i, v in enumerate(nums):
            n = target - v
            if n in o:
                return [o[n], i]
            else:
                o[v] = i


class Solution:
    def twoSum(self, nums, target: int):
        o = {}
        for i in range(len(nums)):
            n = nums[i]
            o[n] = i

        for j in range(len(nums)):
            n1 = nums[j]
            n2 = target - n1
            if n2 in o:
                return [j, o[n2]]


a = [2, 7, 11, 15]
s = Solution()
r = s.twoSum(a, 9)
print(r)
# @lc code=end

File: 1.两数之和.js
/*
 * @lc app=leetcode.cn id=1 lang=javascript
 *
 * [1] 两数之和
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

var twoSum = function (nums, target) {
  for (let i = 0; i < nums.length; i++) {
    for (let j = i + 1; j < nums.length; j++) {
      const n1 = nums[i];
      const n2 = nums[j];
      if (n1 + n2 == target) {
        return [i, j]
      }
    }
  }
};
var twoSum = function (nums, target) {
  let o = {}
  for (let i = 0; i < nums.length; i++) {
    let n = nums[i]
    let n2 = target - n
    if (n2 in o) {
      return [i, o[n2]]
    } else {
      o[n] = i
    }
  }
};
// @lc code=end

